File: Patterns/Manuscript/manuscript/code/java/DesignPatterns/src/main/java/factorymethod/ParkVisitor.java
package factorymethod;

class ParkVisitor {
    public static void main(String[] args) {
        FoodStall pizzaStall = new PizzaStall();
        FoodStall iceCreamStall = new IceCreamStall();

        // Ordering dishes from different food stalls
        Product pizza = pizzaStall.takeOrder();
        Product iceCream = iceCreamStall.takeOrder();
    }
}

File: code/python/factorymethod/ParkVisitor.py
from PizzaStall import PizzaStall
from IceCreamStall import IceCreamStall

if __name__ == "__main__":
    pizza_stall = PizzaStall()
    ice_cream_stall = IceCreamStall()

    pizza = pizza_stall.take_order()
    ice_cream = ice_cream_stall.take_order()
